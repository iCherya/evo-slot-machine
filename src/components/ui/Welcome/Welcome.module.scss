@import '@/styles/variables.scss';

.wrapper {
  display: flex;
  flex-direction: column;
  align-items: center;
  position: relative;
  background-color: $color-gold;
  border-radius: $spacing-base;
  padding: $spacing-base;
  // border: none;
}

.logoWrapper {
  border-radius: $spacing-base;
  overflow: hidden;
  max-height: 600px;
  display: flex;
  align-items: center;
  justify-content: center;
}

.buttonWrapper {
  position: absolute;
  top: 75%;
  left: 75%;
  transform: translate(-50%, -50%);
}
