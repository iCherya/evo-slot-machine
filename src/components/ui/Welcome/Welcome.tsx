import React, { useCallback } from 'react';
import { observer } from 'mobx-react-lite';

import { useStore } from '@/store';
import { TranslateText } from '@/components/logic/Translations';
import { Button } from '@/components/ui/Button';

import welcomeBanner from '/assets/welcome-banner.jpeg';
import styles from './Welcome.module.scss';

export const Welcome: React.FC = observer(() => {
  const { game } = useStore();

  const onClickHandler = useCallback(() => {
    game.startGame();
  }, [game]);

  return (
    <div className={styles.welcome}>
      <div className={styles.banner}>
        <img src={welcomeBanner} alt="Lucky Spin Banner" />
      </div>
      <div className={styles.button}>
        <Button action={onClickHandler} type="secondary">
          <TranslateText translationKey="ui.welcome.play.button.text" />
        </Button>
      </div>
    </div>
  );
});
