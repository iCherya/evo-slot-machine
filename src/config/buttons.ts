export const BUTTON_OPTIONS = {
  SOCIAL: 'social',
  DEFAULT: 'default',
};
