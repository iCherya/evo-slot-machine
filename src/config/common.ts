export const COMMON = {
  buttonOptions: { SOCIAL: 'social', DEFAULT: 'default' },
  social: [
    { id: 0, type: 'github', link: 'https://github.com/icherya' },
    { id: 1, type: 'linkedin', link: 'https://www.linkedin.com/in/icherya/' },
  ],
};
