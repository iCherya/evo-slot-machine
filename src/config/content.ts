export const CONTENT = {
  header: {
    title: 'Jumbo Slot',
  },
};
