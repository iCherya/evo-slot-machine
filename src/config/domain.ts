export const DOMAIN = {
  currency: 'â‚¬',
  depositValue: 100,
  betValue: 10,
  rowWinMultiplier: 2,
  reelsCount: 3,
  turnoverCount: 10,
  turnoverDuration: 200,
  slotsConfig: [
    { id: 1, price: 4, image: 'assets/slots/slot-1-512.png', name: 'cherry', isWin: false },
    { id: 2, price: 8, image: 'assets/slots/slot-2-512.png', name: 'dollar', isWin: false },
    { id: 3, price: 16, image: 'assets/slots/slot-3-512.png', name: 'seven', isWin: false },
    { id: 4, price: 32, image: 'assets/slots/slot-4-512.png', name: 'grape', isWin: false },
    { id: 5, price: 10, image: 'assets/slots/slot-5-512.png', name: 'bar', isWin: false },
  ],
};
