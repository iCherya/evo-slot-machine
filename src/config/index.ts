export { DOMAIN } from './domain';
export { COMMON } from './common';
