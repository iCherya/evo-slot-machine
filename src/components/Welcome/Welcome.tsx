import React, { useCallback } from 'react';
import { observer } from 'mobx-react-lite';

import { useStore } from '@/store';
import { CONTENT } from '@/config/content';
import { Button } from '@/components/shared/Button';

import welcomeBanner from '../../../public/assets/welcome-banner.jpeg';

import styles from './Welcome.module.scss';

export const Welcome: React.FC = observer(() => {
  const { ui } = CONTENT;
  const { app } = useStore();

  const onClickHandler = useCallback(() => {
    app.startGame();
  }, [app]);

  return (
    <div className={styles.wrapper}>
      <div className={styles.logoWrapper}>
        <img src={welcomeBanner} alt="Lucky Spin Banner" />
      </div>
      <div className={styles.buttonWrapper}>
        <Button action={onClickHandler} type="secondary">
          {String(ui.welcome.buttonText)}
        </Button>
      </div>
    </div>
  );
});
