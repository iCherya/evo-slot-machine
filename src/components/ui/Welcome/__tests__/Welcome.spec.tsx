import { render, fireEvent } from '@testing-library/react';

import { Welcome } from '@/components/ui/Welcome';
import { CONTENT } from '@/config';

const mockedStartGame = jest.fn();

jest.mock('@/store', () => ({
  useStore: jest.fn(() => ({
    game: {
      startGame: mockedStartGame,
    },
  })),
}));

jest.mock('@/components/ui/Button', () => ({
  Button: jest.fn(({ children, action, ...props }) => (
    // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
    <button onClick={action} {...props}>
      {children}
    </button>
  )),
}));

describe('Welcome', () => {
  it('should render main welcome banner', () => {
    const { getByAltText } = render(<Welcome />);
    const welcomeBanner = getByAltText('Lucky Spin Banner');

    expect(welcomeBanner).toBeInTheDocument();
    expect(welcomeBanner).toMatchSnapshot();
  });

  it('should render starting game button', () => {
    const { getByText } = render(<Welcome />);
    const startGameButton = getByText(CONTENT.ui.welcome.buttonText);

    expect(startGameButton).toBeInTheDocument();
    expect(startGameButton).toMatchSnapshot();
  });

  it('should start the game when button clicked', () => {
    const { container, getByRole } = render(<Welcome />);
    const startGameComponent = getByRole('button');

    fireEvent(startGameComponent, new MouseEvent('click', { bubbles: true }));

    expect(mockedStartGame).toHaveBeenCalledTimes(1);
    expect(container).toMatchSnapshot();
  });
});
