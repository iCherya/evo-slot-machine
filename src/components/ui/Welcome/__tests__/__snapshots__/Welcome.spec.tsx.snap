// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Welcome should render main welcome banner 1`] = `
<img
  alt="Lucky Spin Banner"
/>
`;

exports[`Welcome should render starting game button 1`] = `
<span>
  ui.welcome.play.button.text
</span>
`;

exports[`Welcome should start the game when button clicked 1`] = `
<div>
  <div>
    <div>
      <img
        alt="Lucky Spin Banner"
      />
    </div>
    <div>
      <button
        type="secondary"
      >
        <span>
          ui.welcome.play.button.text
        </span>
      </button>
    </div>
  </div>
</div>
`;
